### PR TITLE
Cleanup TPM code

### PR DESCRIPTION
**Pull request by JayKickliter**
This PR contains what _should be_ cosmetic and safety improvements for the TPM feature.

### CODE REVIEW

File: src/tpm/tpm_wrapper.rs [hunk 1]
```diff
@@ -1,240 +1,204 @@
...
+            Fapi_Finalize(&mut self.0);
         }
-
-        Esys_Finalize(&mut *esys_ctx);
     }
+}
 
-    Ok(())
+lazy_static! {
+    static ref TPM_CTX: Mutex<FapiContext> = Mutex::new(FapiContext::new().unwrap());
```
Comment (JayKickliter) [inline src/tpm/tpm_wrapper.rs:55]: TODO: I'd want to get rid of this `unwrap()`, but working with a global `Result<Mutex>`/`Mutex<Result>` was slowing progress.

File: src/tpm/tpm_wrapper.rs [hunk 2]
```diff
@@ -86,78 +60,77 @@ pub fn public_key(key_path: &str) -> Result<Vec<u8>> {
...
-            libc::free(esys_blob as *mut c_void)
-        })?;
+        ))?;
 
+        // TODO: what exactly does this mean?
+        //
+        // > Once the application finishes its direct interaction with
+        // > the TPM it SHALL delete its pointer to the TCTI context
+        //
+        // Are we responsible for freeing `tcti_ctx`?
```
Comment (isergieienkov) [inline src/tpm/tpm_wrapper.rs:94]: nope. GetTcti doesn't allocate new tcti context. It returns pointer to inner structure of TPM context. no need to free it.

File: src/tpm/tpm_wrapper.rs [hunk 3]
```diff
@@ -178,63 +178,52 @@ pub fn ecdh(x: &[u8], y: &[u8], key_path: &str) -> Result<Vec<u8>> {
...
-            return Err(TpmError::tpm_error(
-                "Tss2_MU_TPMS_CONTEXT_Unmarshal",
-                result,
-            ));
-        }
+        ), always => {
+            libc::free(esys_blob as *mut c_void);
+        })?;
+
+        // let mut result;
```
Comment (isergieienkov) [inline src/tpm/tpm_wrapper.rs]: unneeded comment?

Comment (JayKickliter) [reply src/tpm/tpm_wrapper.rs]: yes, but somehow you're looking at an older commit? I'm not sure what happened. Perhaps you started a review long before submitting?

**Review by isergieienkov — DISMISSED** [review]

**Pull request merged**